doc/developer/design/20250226_postgres_style_explain.md

```diff
@@ -0,0 +1,338 @@
+                          Index Cond: (l_orderkey = orders.o_orderkey)
+                          Filter: (l_shipdate > '1995-03-15'::date)
+(16 rows)
+```
+
+Materialize `EXPLAIN`:
+
+```
+Finish order_by=[#1{sum} desc nulls_first, #2{o_orderdate} asc nulls_last] output=[#0..=#3]
+  Project (#0{o_orderkey}, #3{sum}, #1{o_orderdate}, #2{o_shippriority}) (columns=4)
+    Reduce group_by=[#0{o_orderkey}..=#2{o_shippriority}] aggregates=[sum((#3{l_extendedprice} * (1 - #4{l_discount})))] (columns=4)
+      Project (#8{o_orderkey}, #12{o_orderdate}, #15{o_shippriority}, #22{l_extendedprice}, #23{l_discount}) (columns=5)
+        Filter (#6{c_mktsegment} = "BUILDING") AND (#12{o_orderdate} < 1995-03-15) AND (#27{l_shipdate} > 1995-03-15) (columns=33)
+          Join on=(#0{c_custkey} = #9{o_custkey} AND #8{o_orderkey} = #17{l_orderkey}) type=delta (columns=33)
+            implementation
+              %0:customer » %1:orders[#1]KAif » %2:lineitem[#0]KAif
+              %1:orders » %0:customer[#0]KAef » %2:lineitem[#0]KAif
+              %2:lineitem » %1:orders[#0]KAif » %0:customer[#0]KAef
+            ArrangeBy keys=[[#0{c_custkey}]] (columns=8)
+              ReadIndex on=customer pk_customer_custkey=[delta join 1st input (full scan)] (columns=8)
+            ArrangeBy keys=[[#0{o_orderkey}], [#1{o_custkey}]] (columns=9)
+              ReadIndex on=orders pk_orders_orderkey=[delta join lookup] fk_orders_custkey=[delta join lookup] (columns=9)
+            ArrangeBy keys=[[#0{l_orderkey}]] (columns=16)
+              ReadIndex on=lineitem fk_lineitem_orderkey=[delta join lookup] (columns=16)
+
+Used Indexes:
+  - materialize.public.pk_customer_custkey (delta join 1st input (full scan))
+  - materialize.public.pk_orders_orderkey (delta join lookup)
+  - materialize.public.fk_orders_custkey (delta join lookup)
+  - materialize.public.fk_lineitem_orderkey (delta join lookup)
+
+Target cluster: quickstart
+```
+
+New Materialize `EXPLAIN`:
+
+```
+Finish
+  Order by: sum desc nulls_first, o_orderdate
+  -> Project (columns=4)
+     Columns: o_orderkey, sum, o_orderdate, o_shippriority
+     -> Reduce (columns=4)
+        Group key: o_orderkey..=#2o_shippriority
+        Aggregates: sum((l_extendedprice * (1 - l_discount)))
+        -> Project (columns=5)
+           Columns: o_orderkey, o_orderdate, o_shippriority, l_extendedprice, l_discount
+           -> Filter (columns=33)
+              Predicates: (c_mktsegment = "BUILDING") AND (o_orderdate < 1995-03-15) AND (l_shipdate > 1995-03-15)
+              -> Delta Join (columns=33)
+                 Conditions: c_custkey = o_custkey AND o_orderkey = l_orderkey
+                 Pipelines:
+                   %0:customer » %1:orders[#1]KAif » %2:lineitem[#0]KAif
+                   %1:orders » %0:customer[#0]KAef » %2:lineitem[#0]KAif
+                   %2:lineitem » %1:orders[#0]KAif » %0:customer[#0]KAef
```

Comment:
We can probably leave these off by default, unless `WITH (JOIN IMPLEMENTATIONS)` is specified.